client/client.go

```diff
@@ -324,7 +325,26 @@ func (c *Client) SetServers(servers []string) {
 	if servers == nil {
 		servers = make([]string, 0)
 	}
-	c.servers = servers
+	// net.ResolveTCPAddr requires port to be set, if one is not provided, supply default port
+	// Using net.SplitHostPort in the event of IPv6 addresses with multiple colons.
+	// IPv6 addresses must be passed in with brackets,
+	// i.e: [::1]:4647 or [::1]
+	setServers := make([]string, len(servers))
+	copy(setServers, servers)
+	for i := 0; i < len(setServers); i++ {
+		if _, _, err := net.SplitHostPort(setServers[i]); err != nil {
+			// multiple errors can be returned here, only searching for missing
+			if strings.Contains(err.Error(), "missing port") {
+				c.logger.Printf("[WARN] client: port not specified, using default port")
+				setServers[i] = fmt.Sprintf("%s:4647", setServers[i])
```

Comment:
I'd probably just use `net.JoinHostPort` here, then I think we can omit `strings` too